packages/nextjs/src/utils/instrumentServer.ts

```diff
@@ -182,40 +198,81 @@ function makeWrappedReqHandler(origReqHandler: ReqHandler): WrappedReqHandler {
     // local.on('error', Sentry.captureException);
 
     local.run(() => {
-      // We only want to record page and API requests
-      if (hasTracingEnabled() && shouldTraceRequest(req.url, publicDirFiles)) {
-        const transaction = Sentry.startTransaction({
-          name: `${(req.method || 'GET').toUpperCase()} ${req.url}`,
-          op: 'http.server',
-        });
-        Sentry.getCurrentHub()
-          .getScope()
-          ?.setSpan(transaction);
-
-        res.__sentry__ = { transaction };
-
-        res.once('finish', () => {
-          const transaction = res.__sentry__?.transaction;
-          if (transaction) {
-            // Push `transaction.finish` to the next event loop so open spans have a chance to finish before the transaction
-            // closes
-            setImmediate(() => {
-              // TODO
-              // addExpressReqToTransaction(transaction, req);
+      const currentScope = Sentry.getCurrentHub().getScope();
+
+      if (currentScope) {
+        currentScope.addEventProcessor(event => addRequestDataToEvent(event, req));
+
+        // We only want to record page and API requests
+        if (hasTracingEnabled() && shouldTraceRequest(req.url, publicDirFiles)) {
+          // pull off query string, if any
+          const reqPath = req.url.split('?')[0];
+
+          // requests for pages will only ever be GET requests, so don't bother to include the method in the transaction
+          // name; requests to API routes could be GET, POST, PUT, etc, so do include it there
+          const namePrefix = req.url.startsWith('/api') ? `${(req.method || 'GET').toUpperCase()} ` : '';
```

Comment:
@HazAT pinging Daniel to give his thoughts on the name prefix for transaction

Reply:
This is what we agreed to in our meeting, no? API requests might have different methods, but page requests are always `GET`, so include the method for API calls, and not for the page ones. I believe I even confirmed it with him afterwards.